src/Language/AST/NamedTypeNode.php

```diff
@@ -9,6 +9,6 @@ class NamedTypeNode extends Node implements TypeNode
     /** @var string */
     public $kind = NodeKind::NAMED_TYPE;
 
-    /** @var NameNode */
+    /** @var NameNode|string */
```

Comment:
I went through usages I found and don't think this can be string. Are you sure?

Maybe it just needs fixing in `Printer`:

```diff
NodeKind::NAMED_TYPE => static function (NamedTypeNode $node) {
-    return $node->name;
+    return $node->name->value;
},
```

Reply:
(hopefully phpstan will help us sort this out--I'm sure there are related items in the baseline file)

Reply:
For the record, found this in the reference library: https://github.com/somashekharsk/AngularTypings/blob/24111c927c09e183f5d9adec73abfd097c76fbc5/types/graphql/language/ast.d.ts#L328

The type of `name` is definitely a `NameNode`, full stop. I don't know what's going on with ours, but I still suspect our `visit` implementation (I compared it with the one in the reference library, and we have diverged quite a bit).

Reply:
I downloaded the javascript reference library, and ran the same test (`PrinterText.testDoesn'tAlterAST`) that I was working with in PHP in the debugger. Take a look at this screenshot from `visit`:

![image](https://user-images.githubusercontent.com/773172/79938950-c786cf00-8412-11ea-8d15-82cec4459f0e.png)

You can see that--just as we see in graphql-php--the complex `name` member is about to be steamrolled with a simple string.

So, assuming there are no bugs in the reference library, I think can can conclude a few things:

1. `NamedTypeNode.name` can be either a string or a `NameNode`
2. The typescript and flow definitions cheat and annotate it a `NameNode` (as we did).
3. PHPStan won't let us get away with that; we have to annotate it as a `string|NameNode` union.

Whew. Any other comments?

Reply:
https://github.com/webonyx/graphql-php/issues/655